libethash/io_posix.c

```diff
@@ -86,9 +90,22 @@ bool ethash_file_size(FILE* f, size_t* ret_size)
 
 bool ethash_get_default_dirname(char* strbuf, size_t buffsize)
 {
+#if defined(__APPLE__)
+	static const char dir_suffix[] = "Ethash/";
+	strbuf[0] = '\0';
+	char* home_dir = getenv("HOME");
+	if (!home_dir || strlen(home_dir) == 0)
```

Comment:
Side Note: I'll create final PR for this on ethash monday morningish. (this being one of the changes to be made)

Regarding macos. Using .* folders is pretty much deprecated in favour of things like ~/Library/Application Support and plist Preferences. We could just put everything in .* folders like in Linux but then AlethZero would be in a separate path -- one or the other had to be used and for a few months now the codebase was putting everything in ~/Library/Application Support. (Fwiw, the plists are sort of like windows registry)

Re: POSIX and such. Apple also has a sandboxing system which it can run apps in. When it does, the app can't access a home directory, it can only access application data. As said before, for macos, getpwuid actually needs to be replaced with the appropriate CF methods.

For now I'll be changing it back to .ethash because I for the 'Library/Application Support' prefix.